specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/integrations.json

```diff
@@ -0,0 +1,653 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Microsoft Defender for Cloud",
+    "description": "API spec for Microsoft.Security (Microsoft Defender for Cloud) resource provider",
+    "version": "2023-07-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/{scope}/providers/Microsoft.Security/integrations": {
+      "get": {
+        "x-ms-examples": {
+          "List integrations by subscription scope": {
+            "$ref": "./examples/integrations/ListBySubscriptionIntegrations_example.json"
+          },
+          "List integrations by security connector scope": {
+            "$ref": "./examples/integrations/ListBySecurityConnectorIntegrations_example.json"
+          },
+          "List integrations by management group scope": {
+            "$ref": "./examples/integrations/ListByManagementGroupIntegrations_example.json"
+          }
+        },
+        "tags": [
+          "Integrations"
+        ],
+        "description": "Get a list of all relevant integrations over a scope",
+        "operationId": "Integrations_List",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "#/parameters/Scope"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/IntegrationsList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/{scope}/providers/Microsoft.Security/integrationsMetadata": {
```

Comment:
>integrationsMetadata [](http://example.com/codeflow?start=43&length=20)

This is not a valid resource type. The resource type should always be plural. If you want to just return metadata object here, consider using POST Action instead of GET API, as metadata doesn't look like an ARM Resource here.

POST /{scope}/providers/Microsoft.Security/getintergationsMetadata

Please note that POST Action should always be a verb, hence, the prefix "get" in the action name "getintergationsMetadata".

Reply:
The "ServiceNowTablesConfigurationsResponse" is still not a valid resource. Implementing GET Collection API requires the response to be in ARM Resource format. If the response needs to be a flexible schema, in other words, this is not actually an ARM resource, then use POST action, instead of GET.

Reply:
changed to post

Reply:
This is still not valid operation. Consider something like this:

POST /{scope}/providers/Microsoft.Security/getintergationsMetadata

Please note that POST Action should always be a verb, hence, the prefix "get" in the action name "getintergationsMetadata".